File: backend/dataset/maxPathSum/maxPathSum_13.py
class Solution:
    def nodeMax(self, node):
        if not node:
            return 0
        else:
            lmax, rmax = self.nodeMax(node.left), self.nodeMax(node.right)
            self.res = max(self.res, node.val + lmax + rmax)
            return max(node.val + max(lmax, rmax), 0)
    def maxPathSum(self, root: TreeNode) -> int:
        self.res = root.val
        self.nodeMax(root)
        return self.res

File: backend/CDataset/maxPathSum/maxPathSum_13.cpp
int f;
int Function(TreeNode* root){
if(root==NULL){
return 0;
}
int a=Function(root->left);
int b=Function(root->right);
int k=max(a,b);
k=max(k,0);
k=k+root->val;
f=max(f,k);
f=max(f,root->val+a+b);

    return k;
    
}

int maxPathSum(TreeNode* root) {
    f=INT_MIN;
    Function(root);
    
    return f;
    
}